lib/dfdutils/printdfd.c

```diff
@@ -59,39 +735,176 @@ void printDFD(uint32_t *DFD)
            KHR_DFDVAL(BDB, BYTESPLANE6),
            KHR_DFDVAL(BDB, BYTESPLANE7));
     for (sample = 0; sample < samples; ++sample) {
-        int channelId = KHR_DFDSVAL(BDB, sample, CHANNELID);
-        printf("    Sample %d\n", sample);
-        printf("Qualifiers %x", KHR_DFDSVAL(BDB, sample, QUALIFIERS) >> 4);
-        printf(" Channel 0x%x", channelId);
-        if (model == KHR_DF_MODEL_UASTC) {
-            printf(" (%s)",
-                   channelId == KHR_DF_CHANNEL_UASTC_RRRG ? "RRRG"
-                 : channelId == KHR_DF_CHANNEL_UASTC_RGBA ? "RGBA"
-                 : channelId == KHR_DF_CHANNEL_UASTC_RRR ? "RRR"
-                 : channelId == KHR_DF_CHANNEL_UASTC_RGB ? "RGB"
-                 : channelId == KHR_DF_CHANNEL_UASTC_RG ? "RG"
-                 : "unknown");
-        } else if (model == KHR_DF_MODEL_ETC1S) {
-            printf(" (%s)",
-                   channelId == KHR_DF_CHANNEL_ETC1S_AAA ? "AAA"
-                 : channelId == KHR_DF_CHANNEL_ETC1S_GGG ? "GGG"
-                 : channelId == KHR_DF_CHANNEL_ETC1S_RRR ? "RRR"
-                 : channelId == KHR_DF_CHANNEL_ETC1S_RGB ? "RGB"
-                 : "unknown");
-        } else {
-            printf(" (%c)",
-               "RGB3456789abcdeA"[channelId]);
-        }
-        printf(" Length %d bits Offset %d\n",
+        khr_df_model_channels_e channelId = KHR_DFDSVAL(BDB, sample, CHANNELID);
+        printf("Sample %d:\n", sample);
+
+        khr_df_sample_datatype_qualifiers_e qualifiers = KHR_DFDSVAL(BDB, sample, QUALIFIERS) >> 4;
+        printf("    Qualifiers: 0x%x (", qualifiers);
+        printFlagBits(qualifiers, dfdToStringSampleDatatypeQualifiers);
+        printf(")\n");
+        printf("    Channel ID: 0x%x", channelId);
```

Comment:
This should be Channel Type. That is the term the DF spec uses. I'm sorry for my previous comment. The macros in `khr_df.h` for accessing that field use `CHANNELID` hence my erroneous request. Probably we should add `CHANNELTYPE` aliases to `khr_df.h` for the `CHANNELID` macros.

Reply:
I will change it and make an alias for it too. And I presume the JSON key should also be called channelType.

Reply:
Yes the JSON key should be channelType but it still is, unless I missed a place you previously changed.